packages/back-end/src/api/features/updateFeature.ts

```diff
@@ -0,0 +1,140 @@
+import { updateFeatureValidator } from "../../validators/openapi";
+import {
+  getFeature,
+  updateFeature as updateFeatureToDb,
+} from "../../models/FeatureModel";
+import { getExperimentMapForFeature } from "../../models/ExperimentModel";
+import {
+  addIdsToRules,
+  getApiFeatureObj,
+  getSavedGroupMap,
+  updateInterfaceEnvSettingsFromApiEnvSettings,
+} from "../../services/features";
+import { FeatureInterface } from "../../../types/feature";
+import { getEnabledEnvironments } from "../../util/features";
+import { addTagsDiff } from "../../models/TagModel";
+import { auditDetailsUpdate } from "../../services/audit";
+import { parseJsonSchemaForEnterprise, validateEnvKeys } from "./postFeature";
+
+export const updateFeature = createApiRequestHandler(updateFeatureValidator)(
+  async (req): Promise<UpdateFeatureResponse> => {
+    const feature = await getFeature(req.organization.id, req.params.id);
+    if (!feature) {
+      throw new Error(`Feature id '${req.params.id}' not found.`);
+    }
+
+    const { owner, archived, description, project, tags } = req.body;
+
+    // check permissions for previous project and new one
+    req.checkPermissions("manageFeatures", [
+      feature.project ?? "",
+      project ?? "",
+    ]);
+
+    if (project != null) {
+      req.checkPermissions(
+        "publishFeatures",
+        feature.project,
+        getEnabledEnvironments(feature)
+      );
+      req.checkPermissions(
+        "publishFeatures",
+        project,
+        getEnabledEnvironments(feature)
+      );
+    }
+
+    const orgEnvs = req.organization.settings?.environments || [];
+
+    // ensure environment keys are valid
+    if (req.body.environments != null) {
+      validateEnvKeys(
+        orgEnvs.map((e) => e.id),
+        Object.keys(req.body.environments ?? {})
+      );
+    }
+
+    // ensure default value matches value type
+    let defaultValue;
+    if (req.body.defaultValue != null) {
+      defaultValue = validateFeatureValue(feature, req.body.defaultValue);
+    }
+
+    const environmentSettings =
+      req.body.environments != null
+        ? updateInterfaceEnvSettingsFromApiEnvSettings(
+            feature,
+            req.body.environments
+          )
+        : null;
+
+    const jsonSchema =
+      feature.valueType === "json" && req.body.jsonSchema != null
+        ? parseJsonSchemaForEnterprise(req.organization, req.body.jsonSchema)
+        : null;
+
+    const updates: Partial<FeatureInterface> = {
+      ...(owner != null ? { owner } : {}),
+      ...(archived != null ? { archived } : {}),
+      ...(description != null ? { description } : {}),
+      ...(project != null ? { project } : {}),
+      ...(tags != null ? { tags } : {}),
+      ...(defaultValue != null ? { defaultValue } : {}),
+      ...(environmentSettings != null ? { environmentSettings } : {}),
+      ...(jsonSchema != null ? { jsonSchema } : {}),
+    };
+
+    if (updates.environmentSettings) {
```

Comment:
There are some other fields that require this permission since it can change the published SDK payloads:
- defaultValue
- project
- archived

Reply:
Updated